tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -684,3 +698,80 @@ def sync_volumes(volumes, index, vol_options, sync_ttl):
             b_index += 1
         except KeyError:
             break
+
+
+def brick_status_alert(hostname):
+    # fetching brick details of disconnected node
+    bricks = NS._int.client.read(
+        "clusters/%s/Bricks/all/%s" % (
+            NS.tendrl_context.integration_id,
+            hostname
+        )
+    )
+    for brick_info in bricks.leaves:
+        try:
+            brick_path = brick_info.key
+            lock = None
+            lock = etcd.Lock(
+                NS._int.client,
```

Comment:
Here read lock is used to avoid the rasing same alert for multiple times by different nodes, the problem is if all the nodes raising the brick alert at the same time then  node_agent in different nodes will raise the same alert.

If we use sleep rather than lock then peer disconnect alert will be raised by different node because it is executed in thread.

Reply:
So the read lock will solve the problem

Reply:
yes